rebar.config

```diff
@@ -10,8 +10,9 @@
 
 {profiles, [{ test, [
                      {deps, [meck]},
-                     {erl_opts, [nowarn_export_all]}
+                     {erl_opts, [nowarn_export_all, nowarn_deprecated_function]}
                     ]
             }]}.
 
-{erl_opts, [debug_info, warnings_as_errors]}.
+%{erl_opts, [debug_info, warnings_as_errors]}.
```

Comment:
Can we fix the warnings we're not exposing with this?

Reply:
The warnings are fixed in the sigv4 update (coming "soon"?), so my default was to just ignore them here to avoid duplicating work.  Would you like me to fix the warnings here?

Reply:
as long as we have a ticket | story in the sig_v4 epic to reconfirm that warnings_as_errors is enabled this should be fine. 

An observation - making the changes for sig_v4 in mini_s3 would mean we need to create and merge the PR in mini_s3 before we merge the work for sig_v4.

Feel free to decide which one is easier.

Reply:
@PrajaktaPurohit @marcparadise I went with a solution of copying the deprecated modules over from erlang/otp.  If this is deemed acceptable, I will make issues to remove these files after the upgrade to sigv4, when they will no longer be needed (this entire deprecation issues goes away with that update).

Reply:
Successfully compiles locally, and passes eunit tests.

Reply:
```
mac:mini_s3 lbaker$ ./rebar3 eunit
===> Verifying dependencies...
===> Fetching meck v0.9.2
===> Analyzing applications...
===> Compiling meck
===> Analyzing applications...
===> Compiling mini_s3
===> Performing EUnit tests...
................................
Finished in 4.675 seconds
32 tests, 0 failures
mac:mini_s3 lbaker$ ./rebar3 ct
===> Verifying dependencies...
===> Analyzing applications...
===> Compiling mini_s3
===> Running Common Test suites...
mac:mini_s3 lbaker$
```